Repository/UserRepository.cs

```diff
@@ -0,0 +1,71 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Prac_assignments.Model;
+using Prac_assignments.Model.Dto;
+using Prac_assignments.TenantDbContext;
+
+namespace Prac_assignments.Repository
+{
+    public class UserRepository : IUserRepository
+    {
+        private readonly IMapper _mapper;
+        private readonly TenantDBContext _dbContext;
+        public UserRepository(IMapper mapper, TenantDBContext dBContext)
+        {
+            _mapper = mapper;
+            _dbContext = dBContext;
+        }
+        public async Task<UserDto> CreateUser(UserDto UserDto)
+        {
+            var user = _mapper.Map<UserProfile>(UserDto);
+            _dbContext.UserProfile.Add(user);
+            await _dbContext.SaveChangesAsync();
+            var newUser = _mapper.Map<UserDto>(user);
+            return newUser;
+        }
+
+        public async Task<bool> DeleteUser(int id)
+        {
+            if (id == 0)
+            {
+                throw new ArgumentException("User Id is null or not assign value");
```

Comment:
what if the Id passed is 100 and you only have 20 users?

Reply:
So you are saying I need to put one more condition where I have to check user ID should not be greater than the total users 
or should I change the message for Exception?